File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    for i in range(0, len(queries)):
        temp = []
        for j in range(queries[i][0],queries[i][1]+1):
            if arr[j] > queries[i][2]:
                temp.append(arr[j])
        if len(temp) == 0:
            answer.append(-1)
        else:
            temp.sort()
            answer.append(temp[0])
    return answer

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    vector<int> answer;
    for (auto query : queries) {
        int s = query[0];
        int e = query[1];
        int k = query[2];

        int minValue = 1000000;

        for (int i = s; i <= e; i++) {
            if (arr[i] > k) {
                minValue = min(minValue, arr[i]);
            }
        }

        if (minValue == 1000000)
            answer.push_back(-1);
        else
            answer.push_back(minValue);
    }
    return answer;
}